lib/wificom/wifi_nina.py

```diff
@@ -30,36 +28,28 @@ def __init__(self, esp32_cs, esp32_busy, esp32_reset, esp32_sck, esp32_mosi, esp
 		self.esp = adafruit_esp32spi.ESP_SPIcontrol(self.spi, self.esp32_cs, self.esp32_ready, \
 			 self.esp32_reset)
 		socket.set_interface(self.esp)
-
+	# pylint: disable=unused-argument,invalid-name,inconsistent-return-statements
 	def connect(self):
 		'''
 		Connect to a supported board's WiFi network
 		'''
-		print("Connecting to WiFi network [" + secrets_wifi_ssid + "]...")
-
-		secrets_payload = {
-			'ssid': secrets_wifi_ssid,
-			'password': secrets_wifi_password
-		}
-
-		wifi_manager = adafruit_esp32spi_wifimanager.ESPSPI_WiFiManager(self.esp, secrets_payload)
-
-		# Check status, attempt reconnect if connect attempt fails
-		try:
-			esp_status = self.esp.status
-			if esp_status == 0:
-				wifi_manager.connect()
-		except RuntimeError:
-			wifi_manager.reset()
-
-		print("Connected to WiFi network!")
-
-		mqtt_client = MQTT.MQTT(
-			broker=secrets_mqtt_broker,
-			username=secrets_mqtt_username.lower(),
-			password=secrets_mqtt_password,
-		)
-
-		MQTT.set_socket(socket, self.esp)
-
-		return mqtt_client
+		for network in secrets_wireless_networks:
+			#pylint: disable=unused-variable
+			for i in range(3):
+				try:
+					print("Connecting to", network['ssid'])
+					self.esp.connect_AP(network['ssid'], network['password'])
+
+					mqtt_client = MQTT.MQTT(
+						broker=secrets_mqtt_broker,
+						username=secrets_mqtt_username.lower(),
+						password=secrets_mqtt_password,
+					)
+
+					MQTT.set_socket(socket, self.esp)
+
+					return mqtt_client
+				#pylint: disable=broad-except,invalid-name
+				except Exception as e:
+					print("Failed to connect: ", e)
```

Comment:
print(e) doesn't show the type of the exception, which we might want to know about.

Reply:
Added a new method to display the error type also, is this what you're looking for?